In [9]:
from google.colab import files
uploaded = files.upload()

Saving coding_potential.csv to coding_potential (1).csv


In [0]:
import io
import pandas as pd
df = pd.read_csv(io.BytesIO(uploaded['coding_potential.csv']))



In [11]:
df

#ID  ORF-integrity  ORF-coverage  ...         A         T  label
0      0.162071            1.0      0.114583  ...  0.247959  0.244019      1
1      0.431156            1.0      0.086155  ...  0.210456  0.262064      1
2      0.874798            1.0      0.081715  ...  0.219887  0.247373      1
3      0.853372            1.0      0.081498  ...  0.225238  0.256787      1
4      0.789583            1.0      0.080668  ...  0.220987  0.261293      1
...         ...            ...           ...  ...       ...       ...    ...
57518  0.000000            0.0      0.064516  ...  0.138298  0.255319      0
57519  0.200000            1.0      0.048544  ...  0.221154  0.269231      0
57520  0.461538            1.0      0.066667  ...  0.232044  0.292818      0
57521  0.253125            1.0      0.044025  ...  0.216301  0.253918      0
57522  0.382550            1.0      0.054422  ...  0.263514  0.277027      0

[57523 rows x 39 columns]

1. How many rows and columns does the coding potential.csv have?

  57523 rows × 39 columns



In [27]:
X = df.drop('label',1) #features
y = df['label'] #target

print(X.shape) # Viewing the shape of X
print(y.shape) # Viewing the shape of y

X.head()


(57523, 38)
(57523,)


#ID  ORF-integrity  ORF-coverage  ...         G         A         T
0  0.162071            1.0      0.114583  ...  0.353223  0.247959  0.244019
1  0.431156            1.0      0.086155  ...  0.232239  0.210456  0.262064
2  0.874798            1.0      0.081715  ...  0.243331  0.219887  0.247373
3  0.853372            1.0      0.081498  ...  0.222304  0.225238  0.256787
4  0.789583            1.0      0.080668  ...  0.223767  0.220987  0.261293

[5 rows x 38 columns]

In [0]:
from sklearn.model_selection import train_test_split
Xtrain, Xtest, ytrain, ytest = train_test_split(X, y)

In [22]:
print(Xtrain.shape) # Viewing the shape of X
print(ytrain.shape) # Viewing the shape of y

print(Xtest.shape) # Viewing the shape of X
print(ytest.shape) # Viewing the shape of y

(43142, 38)
(43142,)
(14381, 38)
(14381,)


2. When you split the data into segments, how many samples (rows) are in the training and test data?

  Rows in training data: 43142
  Rows in test data: 14381

In [19]:
from sklearn.ensemble import RandomForestClassifier
model_rf = RandomForestClassifier(n_estimators=100, random_state=42)
model_rf.fit(Xtrain, ytrain)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=42, verbose=0,
                       warm_start=False)

3. Report the confusion matrices and classification reports for the training and test data

In [23]:
from sklearn.metrics import confusion_matrix, classification_report

ypred_train = model_rf.predict(Xtrain)
print('Train Confusion Matrix: ')
print(confusion_matrix(ytrain, ypred_train))
print('')

print('Train Classification report:')
print(classification_report(ytrain, ypred_train))
print('')

Train Confusion Matrix: 
[[18164     3]
 [    3 24972]]

Train Classification report:
              precision    recall  f1-score   support

           0       1.00      1.00      1.00     18167
           1       1.00      1.00      1.00     24975

    accuracy                           1.00     43142
   macro avg       1.00      1.00      1.00     43142
weighted avg       1.00      1.00      1.00     43142




In [24]:
ypred = model_rf.predict(Xtest)
print('Test Confusion Matrix: ')
print(confusion_matrix(ytest, ypred))
print('')
print('Test Classification report:')
print(classification_report(ytest, ypred))

Test Confusion Matrix: 
[[5831  165]
 [ 168 8217]]

Test Classification report:
              precision    recall  f1-score   support

           0       0.97      0.97      0.97      5996
           1       0.98      0.98      0.98      8385

    accuracy                           0.98     14381
   macro avg       0.98      0.98      0.98     14381
weighted avg       0.98      0.98      0.98     14381



## 4. Explain what the confusion matrix describes.

     A confusion matrix is a table that is used to describe the performance of a classification model on a set of test data. 
     It allows the visualization of the performance of an algorithm.


## 5. What are precision, recall and F1-score?

      Precision: Percentage of the items that the system labeled as positive and that are in fact positive. 
      Recall :  Percentage of the items actually present in the input that were correctly identified by the system.
      F1-Score: Since we have two measures (Precision and Recall) it helps to have a measurement that represents both of them. We calculate an F-measure which uses Harmonic Mean in place of Arithmetic Mean as it punishes the extreme values more. F1-Score is just an arithmetic mean of precision and recall since beta is 1.

## 6. Does the classifier perform better when predicting the training data or the test data targets?

      Accuracy of training data is 100% and accurarcy of test data is 98%. So it means classifier perform better predicting the training data. But the difference is not too significant. The acccuracy of test data is also quite enough.


## 7. Why is there a difference between the training and test prediction accuracies?

      In a dataset, a training set is implemented to build up a model, while a test (or validation) set is to validate the model built. And the size of the train set is much more than the test set size. In machine learning, the bigger the set size the better train you get. So, training accuracy is more than the test accuracy. But the difference is not too much here in the case. 


##Alper Bingol 23661
